### PR TITLE
support Instagram API JSON Error Response. 

### PR DESCRIPTION
**Pull request by rakd**
@ahmdrz 

When users login failed, Instagram return JSON Error Message with 400, now we are handling it as "logged out" if the response code = 400. 

e.g. in case of login error. the response will be below with code=400. 
```
{"message": "The password you entered is incorrect. Please try again.", "invalid_credentials": true, "error_title": "Incorrect password for wrong_username", "buttons": [{"title": "Try Again", "action": "dismiss"}], "status": "fail", "error_type": "bad_password"}
```

I think we should take care of error message in JSON. 

Thanks

### CODE REVIEW

**Comment by ahmdrz** [comment]
@rakd

Thank you - as soon as possible I will check your PR , 

Good luck :+1:

File: goinsta_test.go [hunk 1]
```diff
@@ -425,10 +425,21 @@ func TestRecentActivity(t *testing.T) {
...
 		t.Fatal(err)
 	}
+
+	t.Logf("recentActivity.Status:%s", recentActivity.Status)
+	t.Logf("recentActivity.ContinuationToken:%d", recentActivity.ContinuationToken)
+	t.Logf("recentActivity.Counts.PhotosOfYou:%d", recentActivity.Counts.PhotosOfYou)
+	t.Logf("recentActivity.Counts.Requests:%d", recentActivity.Counts.Requests)
+	if len(recentActivity.OldStories) > 0 {
+		for _, item := range recentActivity.OldStories {
+			t.Logf("PK=%s, type=%d, text=%s ", item.PK, item.Type, item.Args.Text)
```
Comment (ahmdrz) [inline goinsta_test.go]: My account is a private account for test on Travis.
So please do not share anything of privacy information.

File: goinsta.go [hunk 2]
```diff
@@ -1097,8 +1097,17 @@ func (insta *Instagram) TotalUserFollowers(userID int64) (response.UsersResponse
 	}
 }
 
-func (insta *Instagram) GetRecentActivity() ([]byte, error) {
-	return insta.sendSimpleRequest("news/inbox/?")
+func (insta *Instagram) GetRecentActivity() (response.RecentActivityResponse, error) {
+	result := response.RecentActivityResponse{}
+	bytes, err := insta.sendSimpleRequest("news/inbox/?")
```
Comment (ahmdrz) [inline goinsta.go]: Good , But now we have it on [Updated function of GetRecentActivity](https://github.com/ahmdrz/goinsta/blob/82b8eb0d9a3b687b2be5809fc0946a68d5170875/goinsta.go#L1101)

File: errors.go [hunk 3]
```diff
@@ -1,10 +1,83 @@
...
+	"encoding/json"
+	"fmt"
+	"log"
+)
 
-// ErrNotFound is returned if the request responds with a 404 status code
+// ErrNotFoundMsg is message text when the request responds with a 404 status code
 // i.e a non existent user
-var ErrNotFound = errors.New("The specified data wasn't found.")
+var ErrNotFoundMsg = "The specified data wasn't found."
```
Comment (ahmdrz) [inline errors.go]: Perfect idea. :+1: I like it :)

File: errors.go [hunk 4]
```diff
@@ -67,7 +67,13 @@ func NewIGAPIError(body string, code int64) *IGAPIError {
 
 // IGAPIErrorResponse ...
 type IGAPIErrorResponse struct {
-	Message string `json:"message"`
+	Message            string `json:"message"`
+	InvalidCredentials bool   `json:"invalid_credentials"`
+	ErrorTitle         string `json:"error_title"`
+	Buttons            []struct {
```
Comment (ahmdrz) [inline errors.go]: :+1:

**Comment by ahmdrz** [comment]
Sorry @rakd for my late answering.
Please update your forked with latest changes on main repository of `goinsta`.
And I will happy to see you again , I'll waiting for you :hourglass_flowing_sand:

**Comment by themester** [comment]
Added here: https://godoc.org/gopkg.in/ahmdrz/goinsta.v2

**Pull request closed without merging**